mitodl/ocw-studio
# add website collection APIs

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #344
closes #316 

#### What's this PR do?

This adds APIs for creating and updated WebsiteCollections and also WebsiteCollectionItems. Together these allowing creating and naming collections and then adding websites to them, changing their order, and so on.

#### How should this be manually tested?

You can try things out at `/api/collections` and `/api/collections/$ID/items`. You should be able to do things like:

- POST to `/api/collections` to create a new one
- GET `/api/collections/$ID` to get info about a collection
- PATCH or PUT to `/api/collections/$ID` to change description, title
- GET `/api/collections/$ID/items` to get a list of items
- POST to `/api/colletctions/$ID/items` to add an item to a collection (i.e. to create a new `WebsiteCollectionItem`)
- PATCH or PUT to `/api/collections/$COL_ID/items/$ITEM_ID` to change position and so on

And you should also be able to DELETE things. I think that covers what is here.

One thing I am a little unsure about is the permissions. I implemented some new permissions classes to use with this, but as I am a little unfamiliar with how these work I thought they probably could use a close review.

I also added an `owner` field to the `WebsiteCollection` which is used to record who creates the object (or, well, I guess it could be changed later too if we wanted someone else to own a given collection).
